pacman/executor/algorithm_metadata_xml_reader.py

```diff
@@ -51,45 +48,61 @@ class AlgorithmMetadataXmlReader(object):
 
     __slots__ = [
         # paths to extra metadata files
-        "_xml_paths"
+        "_xml_paths",
+        # current path
+        "_xml_path"
```

Comment:
I dont like making one of many parameters a self object.

Permission to change?

Reply:
I do make use of that assumption (in many places!), and it greatly reduces the number of variables that get passed around explicitly through lots of calls just because some leaf method wants to look something up or make a simple call; such things I think of as “context” and those I like to put hanging off `self` as that's the standard context thing in Python.